packages/client/src/utils/ShoppingCartContext/ShoppingCartContext.jsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  useState,
+  useContext,
+  useMemo,
+  useCallback,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useUserContext } from '../../userContext';
+
+const ShoppingCartContext = createContext({});
+
+export const useShoppingCart = () => {
+  return useContext(ShoppingCartContext);
+};
+
+const ShoppingCartProvider = ({ children }) => {
+  const { user } = useUserContext();
+
+  const [pendingItems, setPendingItems] = useState(() => {
+    const items = localStorage.getItem(user?.uid);
```

Comment:
Maybe you can sync up with @Yuliia-Balandiuk to talk about creating a custom hook for local storage.

Reply:
We have changed mind about ShoppingCartContext, it will be handled with database

Reply:
You'll put the users current shopping cart into the database as they are preparing it?
That doesn't make a lot of sense in my opinion. Imagine if they stop midway before making an order, you'll end up with a lot of junk in the database. 

For this kind of data, local storage is perfect. The actual order should of course end up in the database once it's confirmed and paid for.

Reply:
If we look on web shops which provides best service, they keep cart data in database, Amazon one of them. Space for data is very cheap in our days. So I'm vouting for database cart data.

Reply:
But you are right, we do need to use localstorage for unauthorized users

Reply:
Space is cheap, but it's mostly cheap **at scale**. There is a million and one things Amazon does that do not apply to this application. 

While I can agree with your point of "best service", there is a lot more that goes into that. Amazon has much more complex systems in place to manage that data, such as expiry dates on incomplete orders, not to mention the amount of effort they have put into making their sales funnel streamlined so that a large majority of shopping carts actually get turned into orders. 

It's good to be inspired, but it is critical to consider how your situation is different from that of the big tech companies, at least until you actually work for one of the big tech companies. Amazon doesn't use MySQL to store this kind of data and at the scale your solving for right now, you don't need such complexity, and in some situations you cannot even afford such complexity. Amazon did not start out huge, they grew!